src/services/Sso.php

```diff
@@ -0,0 +1,199 @@
+        echo '<img src='.$forumUrl.'"/register/logout?authtoken='.$_COOKIE['forumLogoutToken'].'" border="0" width="0" height="0" alt="" id="imageTag">';  
+        Websitetoolboxforum::getInstance()->sso->resetCookieOnLogout();  
+      }
+    }
+    function resetCookieOnLogout(){
+      setcookie('forumLogoutToken', 0, time() - 3600, "/");
+      unset($_COOKIE['forumLogoutToken']);
+      setcookie('forumLoginUserid', '', time() - 3600, "/");
+      unset($_COOKIE['forumLoginUserid']);
+      setcookie('loginRemember', '', time() - 3600, "/");
+      unset($_COOKIE['loginRemember']);
+   }
+   function sendRequest($requestData,$requestUrl,$postType){
+      $ch = curl_init();
+      curl_setopt($ch,CURLOPT_URL,$requestUrl);
+      curl_setopt($ch, CURLOPT_POST, 1);
+      curl_setopt($ch,CURLOPT_RETURNTRANSFER,true);
+      if($postType == 'json'){
+        curl_setopt($ch,CURLOPT_POSTFIELDS,json_encode($requestData));
+      }else{
+        curl_setopt($ch,CURLOPT_POSTFIELDS,http_build_query($requestData));
+      }
+      curl_setopt($ch, CURLOPT_HTTPHEADER, array(                                                                          
+      'Content-Type: application/json','Accept: application/json'));      
+      curl_setopt($ch, CURLOPT_HEADER, 0);
+      $response = curl_exec($ch); 
+      $result = json_decode($response, true); 
+      return $result;
+   }
+   function renderJsScriptEmbedded($forumUrl){   
+
+        $js = <<<JS
+  (  
+   function renderEmbeddedHtmlWithAuthtoken()
+  {  var embedUrl  = "{$forumUrl}";    
+    var wtbWrap = document.createElement('div');
+    wtbWrap.id = "wtEmbedCode";
+    var embedScript = document.createElement('script');
+    embedScript.id = "embedded_forum";
+    embedScript.type = 'text/javascript';
+    var wtbToken = getCookie();
+    if(typeof wtbToken != 'undefined' && wtbToken){
+      embedUrl += "/js/mb/embed.js?authtoken="+wtbToken;
+    } 
+    embedScript.src = embedUrl; 
+    wtbWrap.appendChild(embedScript); 
+    document.getElementById('embedForum').appendChild(embedScript);
+
+  })();
+JS;
+      return $js ;
+    }
+   
+function renderJsScriptUnEmbedded() 
+    {   
+
+        $js = <<<JS
+        (  
+         function renderEmbeddedUnHtmlWithAuthtoken()
+        { 
+        var links = document.getElementsByTagName('a');
+        for(var i = 0; i< links.length; i++){
+          var str = links[i].href; 
+          for(var j = 0; j< str.length; j++){
+            var res = str.split("."); 
+            if(res[j] == 'websitetoolbox'){ 
+                var linkToChange = links[i]; 
+                var wtbToken = getCookie(); 
+                if(typeof wtbToken != 'undefined' && wtbToken){
+                    linkToChange.setAttribute("href", linkToChange+"?authtoken="+wtbToken);
+                }
+            }
+            
+          }
+        }
+        })();
+JS;
+        return $js ;
+    }
+   
+}
```

Comment:
You have used curl code in two different places in the file. Please check and use it only one place. Create a function for it.